static/js/message_util.js

```diff
@@ -20,7 +20,15 @@ function add_messages(messages, msg_list, opts) {
 exports.add_old_messages = function (messages, msg_list) {
     return add_messages(messages, msg_list, {messages_are_new: false});
 };
+
 exports.add_new_messages = function (messages, msg_list) {
+    if (msg_list === current_msg_list && !msg_list.data.fetch_status.has_found_newest()) {
```

Comment:
Do we need the check on `current_msg_list`?  I would think we don't; we don't want to add the message to non-visible message lists like home_msg_list that might not be caught up, either,